app/models/user.rb

```diff
@@ -416,6 +417,22 @@ def customize_digest(type)
     end
   end
 
+  def generate_token
+    user_id_and_time = [id, Time.now]
```

Comment:
May be we should use `hash` here instead of array ? 
`{ :id => id, :timestamp => Time.now }`

That would be more descriptive than using something like `decrypted_data[0]`.